### ISSUE TITLE
As a volunteer, I would like to complete the 3 month follow-up survey for an expert panel assignment

### ISSUE THREAD

**Issue by yugen** [clingen-data-model/community-curation#13]
For comprehensive volunteers
- [x] Build survey (repo:/Documentation/Source Documents/3 month volunteer survey.docx)
- [x] Schedule 3-month survey for 90 days after assignment to EP
- [x] Send email to user at 90 days from EP asssignment
- [x] Send reminder 1 97 days from EP assignment
- [x] Send reminder 2  111 days from EP assignment
- [x] Wire up end point for volunteers to anonymously take the followup survey

**Comment by yugen** [comment]
@courtneythaxton Reminder that I need copy you'd like to include in the email.  Also, if all this stuff lives in Google Drive somewhere can you invite me and I can dig around for this documentation myself?

Thanks!

**Comment by courtneythaxton** [comment]
hi @yugen I have emailed Liz to find this email and provide you with the draft. Hopefully it will be coming soon.

**Comment by courtneythaxton** [comment]
Hi @yugen Here is the current C3 google doc that may have some items of interest for the future. Although admittedly there may be some that I had not included before this time.

https://drive.google.com/open?id=1MdC-BV4GpYWEj-0wZC6UcpD-4yY1OVj5

**Comment by yugen** [comment]
@courtneythaxton and @lizzkearns, this is ready for you to test.

I've created three test volunteers with expert panel assignments backdated to 2019-12-11, 2019-12-04, and 2019-11-20 (90 days from tomorrow, 97 days from tomorrow, and 111 days from tomorrow.  So when the script runs tonight you should see 3 emails in the logs corresponding to the initial notification, first reminder, and 2nd reminder.

The dummy users are:
|id|name|
|5|Comprehensive Volunteer|
|386|Test Comprehensive|
|387|Tester Testerson|

**To test:**
1. To access the logs visit: https://community-curation.sirsdemo.unc.edu/admin/logs and click on 'mail-2020-03-10.  There you should see three entries, one for each email tot he user.
2. You'll see each email as html, but the text should be correct.
3. If you impersonate any of the users. You can copy/past the URL to the 3-month followup survey into the location bar to take the survey and confirm it reads and behaves as intended.
4. Once you've taken (and finished) the survey as one of those users you should not be able to access it again as that user.  You should be able to access it as an admin.
5. You should see the followup survey data on the volunteer's detail screen under the 'Survey Data' tab.

Let me know if you have any questions.

Thanks,
TJ

**Comment by courtneythaxton** [comment]
Hi @yugen I went to test out this function.

I went to the logs and chose the first instance that indicated emails on 3-10-2020, as there were two. The first option showed 4 entries, 3 of which were targeted to the survey as you mentioned above. 

I then went to the record as an Admin and when impersonaing Comprehensive volunteer. They had an indication of being trained. When I went to the survey data tab, under "application" it indicated the following "This volunteer did not complete an application. Based on the expected workflow this is not possible, but there are a few ways it could have happened:
-This is a test database and you are looking at a volunteer that was created for testing purposes w/o completing an applications survey.
-An admin created a volunteer user using the admin panel.
-Something mysterious is going on and you should contact an administrator."

And for  the "3 month followup" tab, I see "The volunteer hasn't completed a 3month followup yet" for the admin, and "no data yet" when impersonating the volunteer.

**Typo:** Above should there be a space between the 3 and month, to say 3 mont, or 3-month?

**In summary:** I am not able to go to the 3 month survey to complete, and then see how the data is displayed on the volunteers records either by an admin or as the volunteer.

From the larval log of the email I saw this link for the survey: a href=\"http://localhost/volunteer-3-month-followup\">survey</a

I tried this full link which I knew would probably not work, and it did NOT.

I then tried the shorter http://localhost/volunteer-3-month-followup\
I also was returned a "This site can't be reached" error on Google Chrome.

**Comment by yugen** [comment]
@courtneythaxton 
- For the test 'Comprehensive Volunteer' user you should expect to see the notification about them not having completed an application because I am creating them programattically for testing.
- I'm glad you were able to see the emails.  We can check that bit off the list.
- I've fixed the issue with the link to the followup in the email.  To test the survey please go to https://community-curation.sirsdemo.unc.edu/volunteer-three-month while impersonating a volunteer.

**Comment by courtneythaxton** [comment]
Hi @yugen 
I was able to review the survey. 

On the Intro page, can we replace the sentence: "Thank you for volunteering to participate in ClinGen community curation. In order to better understand the needs and experiences of our volunteers, we would greatly appreciate your feedback on this anonymous survey." 
 
**with**

Thank you for volunteering to participate in ClinGen community curation. In order to better understand the needs and experiences of our volunteers, we would greatly appreciate your feedback on this survey. Your answers will be viewable only to yourself and ClinGen Community Curation Working Group administration staff.

**since technically we know the person taking it, so it seems unfair to call it anonymous, do you agree?**

In the survey this sentence is duplicated: Were the training materials required for your specific curation effort understandable and clear?

I was able to view the results of the 3 month survey both from the admin side and the user side.

**Comment by yugen** [comment]
@courtneythaxton I've made changes to the survey that you requested.
